recipes/openscenegraph/all/test_package/conanfile.py

```diff
@@ -0,0 +1,22 @@
+from conans import ConanFile, CMake, CMakeToolchain, tools
+import os
+
+
+class TestPackageConan(ConanFile):
+    settings = "os", "compiler", "build_type", "arch"
+    generators = "cmake_find_package"
+
+    def toolchain(self):
+        toolchain = CMakeToolchain(self)
+        toolchain.definitions["WITH_PNG"] = 1 if self.options["openscenegraph"].with_png else 0
+        toolchain.write_toolchain_files()
+
+    def build(self):
+        cmake = CMake(self)
+        cmake.configure()
+        cmake.build()
+
+    def test(self):
+        if not tools.cross_building(self.settings):
+            bin_path = os.path.join(os.curdir, "test_package") + (".exe" if self.settings.os == "Windows" else "")
```

Comment:
```suggestion
            bin_path = os.path.join("bin", "test_package")
```

Reply:
test_package is not in the bin directory. As far as I can tell, CMakeToolchain doesn't mess with output directories.

Reply:
oops, sorry, didn't see toolchain. Are you sure that extension is required?

Reply:
Actually, it shouldn't. I misread the Windows log.